tests/e2e/vsphere_volume_expansion.go

```diff
@@ -199,8 +215,406 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+
+	volHandle := pv.Spec.CSI.VolumeHandle
+
+	// Wait for PV and PVC to Bind
+	framework.ExpectNoError(fpv.WaitOnPVandPVC(client, namespace, pv, pvc))
+
+	return volHandle, pvc, pv, sc
+}
+
+//createSCwithVolumeExpansionTrueAndDynamicPVC creates storageClass with allowVolumeExpansion set to true and Creates PVC. Waits till PV, PVC are in bound
+func createSCwithVolumeExpansionTrueAndDynamicPVC(f *framework.Framework, client clientset.Interface, namespace string) (string, *v1.PersistentVolumeClaim, *v1.PersistentVolume, *storagev1.StorageClass) {
+	scParameters := make(map[string]string)
+	scParameters[scParamFsType] = ext4FSType
+
+	// Create Storage class and PVC
+	ginkgo.By("Creating Storage Class and PVC with allowVolumeExpansion = true")
+	var storageclass *storagev1.StorageClass
+	var pvclaim *v1.PersistentVolumeClaim
+	var err error
+
+	storageclass, pvclaim, err = createPVCAndStorageClass(client, namespace, nil, scParameters, "", nil, "", true, "")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	// Waiting for PVC to be bound
+	var pvclaims []*v1.PersistentVolumeClaim
+	pvclaims = append(pvclaims, pvclaim)
+	ginkgo.By("Waiting for all claims to be in bound state")
+	persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	pv := persistentvolumes[0]
+	volHandle := pv.Spec.CSI.VolumeHandle
+
+	return volHandle, pvclaims[0], pv, storageclass
+
+}
+
+//createPODandVerifyVolumeMount this method creates POD and verifies VolumeMount
+func createPODandVerifyVolumeMount(f *framework.Framework, client clientset.Interface, namespace string, pvclaim *v1.PersistentVolumeClaim, volHandle string) *v1.Pod {
+	// Create a POD to use this PVC, and verify volume has been attached
+	ginkgo.By("Creating pod to attach PV to the node")
+	pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, execCommand)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	var vmUUID string
+	ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", volHandle, pod.Spec.NodeName))
+	vmUUID = getNodeUUID(client, pod.Spec.NodeName)
+
+	isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volHandle, vmUUID)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gomega.Expect(isDiskAttached).To(gomega.BeTrue(), "Volume is not attached to the node")
+
+	ginkgo.By("Verify the volume is accessible and filesystem type is as expected")
+	_, err = framework.LookForStringInPodExec(namespace, pod.Name, []string{"/bin/cat", "/mnt/volume1/fstype"}, "", time.Minute)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	return pod
+}
+
+//increaseSizeOfPvcAttachedToPod this method increases the PVC size, which is attached to POD
+func increaseSizeOfPvcAttachedToPod(f *framework.Framework, client clientset.Interface, namespace string, originalSizeInMb int64, pvclaim *v1.PersistentVolumeClaim, pod *v1.Pod) {
```

Comment:
I would suggest fetching `originalSizeInMb` in this method itself before resizing, instead of asking the caller of the method to do that

Reply:
made the changes